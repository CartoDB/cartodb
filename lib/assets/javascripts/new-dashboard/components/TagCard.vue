<template>
  <router-link :to="{ name: 'tagSearch', params: { tag: tag.tag } }" class="link">
    <article class="card tag">
      <h3 class="tag__title title is-medium is-semibold">{{ tag.tag }}</h3>
      <ul>
        <li class="tag__count text is-caption">
          {{ $tc('TagCard.maps', tag.maps, { maps: tag.maps }) }}
        </li>
        <li class="tag__count text is-caption">
          {{ $tc('TagCard.datasets', tag.datasets, { datasets: tag.datasets }) }}
        </li>
      </ul>
    </article>
  </router-link>
</template>

<script>
export default {
  name: 'TagCard',
  props: {
    tag: Object
  }
};
</script>

<style lang="scss" scoped>
@import 'stylesheets/new-dashboard/variables';

.card {
  position: relative;
  height: 100%;
  padding: 16px;
  transition: background 300ms cubic-bezier(0.4, 0, 0.2, 1);
  border: 1px solid $light-grey;
  background-color: $white;
}

.tag__title,
.tag__count {
  margin-bottom: 8px;
  color: $text-color;
}

.link:hover {
  text-decoration: none;
}
</style>
