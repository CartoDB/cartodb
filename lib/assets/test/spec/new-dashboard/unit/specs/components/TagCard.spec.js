import { shallowMount } from '@vue/test-utils';
import TagCard from 'new-dashboard/components/TagCard';

const $tc = key => key;

describe('TagCard', () => {
  it('should render properly', () => {
    const tagCardElement = shallowMount(TagCard, {
      propsData: {
        tag: {
          tag: 'Test Tag',
          maps: 1,
          datasets: 3
        }
      },
      mocks: {
        $tc
      }
    });

    expect(tagCardElement).toMatchSnapshot();
  });
});
