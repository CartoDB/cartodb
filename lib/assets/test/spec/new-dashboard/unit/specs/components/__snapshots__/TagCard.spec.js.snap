// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCard should render properly 1`] = `
<router-link to="[object Object]" class="link">
  <article class="card tag">
    <h3 class="tag__title title is-medium is-semibold">Test Tag</h3>
    <ul>
      <li class="tag__count text is-caption">
        TagCard.maps
      </li>
      <li class="tag__count text is-caption">
        TagCard.datasets
      </li>
    </ul>
  </article>
</router-link>
`;
